Simple Search Engine/task/src/search/SearchApp.java

```diff
@@ -0,0 +1,137 @@
+package search;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class SearchApp {
+    private final Scanner scanner;
+    private final SearchIndex searchIndex;
+    private boolean isUpAndRunning;
+
+    public SearchApp() {
+        this.scanner = new Scanner(System.in);
+        this.isUpAndRunning = true;
+        this.searchIndex = new SearchIndex();
+    }
+
+    public String[] askSearchQuery() {
+        System.out.println("Enter a name or email to search all suitable people.");
+        return scanner.nextLine().toLowerCase().split(" ");
+    }
+
+    public String askSearchMethod() {
+        System.out.println("Select a matching strategy: ALL, ANY, NONE");
+        return scanner.nextLine();
+    }
+
+    public boolean getIsUpAndRunning() {
+        return isUpAndRunning;
+    }
+
+    public void runMenuAction() {
+        final String menu = "\n=== Menu ===\n" +
+                "1. Find a person\n" +
+                "2. Print all people\n" +
+                "0. Exit";
+
+        System.out.println(menu);
+        String action = scanner.nextLine();
+
+        switch (action) {
+            case "1":
+                findAPersonByMethod();
+                break;
+            case "2":
+                printDataStore();
+                break;
+            case "0":
+                exitApp();
+                break;
+            default:
+                System.out.println("Incorrect option! Try again.");
+                break;
+        }
+    }
+
+    public void findAPersonByMethod() {
```

Comment:
Let op scopes, maak alleen dingen public die public moeten zijn. Gebruik standaard private, pas als dat niet voldoende is ga je de zichtbaarheid verbreden

Reply:
Done.